src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -452,8 +452,7 @@ InboundLedger::onTimer(bool wasProgress, ScopedLockType&)
 void
 InboundLedger::addPeers()
 {
-    app_.overlay().selectPeers(
-        *this,
+    PeerSet::addPeers(
```

Comment:
This is not moving selectPeers, instead it removes selectPeers and seems to put addPeers in its place. But these functions don't do the same thing. What's the intention for this commit?

Reply:
`selectPeers` was a method (that could have been declared const) on `Overlay` that would take a `PeerSet` by non-const reference and insert peers into it (up to some limit, ordered by a scoring function). Instead, I moved the body of that method to `PeerSet` (which takes `this` as a pointer-to-non-const). I like mutating functions to be defined in the object they mutate. These functions should do the same thing. What is the difference in behavior that you spotted?

Reply:
2 calls to newPeer(), and the lock that are in addPeers() but not selectPeers()

Reply:
I only see one call to `newPeer()` on [line 80 of `PeerSet.cpp`](https://github.com/thejohnfreeman/rippled/blob/e2d06dd496b271c3bbda860d7ca9cd160ed3b23d/src/ripple/overlay/impl/PeerSet.cpp#L80). Where do you see the second?

The lock was transitively acquired by `selectPeers` whenever it called `PeerSet::insert`. `PeerSet::insert` has been inlined into `PeerSet::addPeers`.

Reply:
I just see the one, I C&P'd the function into a note and must have duplicated that call. Is newPeer() new for this function?

Reply:
The call to `newPeer()` is not new. It was [called](https://github.com/thejohnfreeman/rippled/commit/e2d06dd496b271c3bbda860d7ca9cd160ed3b23d#diff-6d55b06f2011cfb72aec36becb4e15c6L57) in `PeerSet::insert()`, which I inlined into `PeerSet::addPeers()`.